tests/draft-next/type.json

```diff
@@ -11,13 +11,48 @@
                 "data": 1,
                 "valid": true
             },
+            {
+                "description": "a negative integer is an integer",
+                "data": -1,
+                "valid": true
+            },
+            {
+                "description": "zero is an integer",
+                "data": 0,
+                "valid": true
+            },
+            {
+                "description": "negative zero is an integer",
+                "data": -0,
+                "valid": true
+            },
             {
                 "description": "a float with zero fractional part is an integer",
                 "data": 1.0,
                 "valid": true
             },
             {
-                "description": "a float is not an integer",
+                "description": "a float with an exponent and a zero fractional part is an integer",
+                "data": 1e0,
+                "valid": true
+            },
+            {
+                "description": "a small float whose non-zero fractional part is truncated in a binary64 float is an integer",
+                "data": 1.00000000000001,
+                "valid": true
+            },
+            {
+                "description": "a large float whose non-zero fractional part is truncated in a binary64 float is an integer",
+                "data": 9007199254740991.5,
+                "valid": true
+            },
```

Comment:
@karenetheridge is correct.  These are not integers.  JSON Schema uses an object model based on the one defined by the JSON specification, RFC 8259, which does not put limits on numeric values.  Range does not matter.  See [Core 6.2](https://json-schema.org/draft/2020-12/json-schema-core#section-6.2) and [Validation 4.2](https://json-schema.org/draft/2020-12/json-schema-validation#section-4.2).

We do already have some tests for [extremely large numbers](https://json-schema.org/draft/2020-12/json-schema-core#section-6.2).  They are in the optional folders as we recognize that they typically are not fully supported and are generally edge-case.  An argument could be made that they should not be optional.